sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c

```diff
@@ -735,33 +768,94 @@ static void set_codec_init_func(const struct snd_soc_acpi_link_adr *link,
 	}
 }
 
+/*
+ * check endpoint status in slaves and gather link ID for all slaves in
+ * the same group to generate different CPU DAI. Now only support
+ * one sdw link with all slaves set with only single group id.
+ */
+static void get_slave_info(const struct snd_soc_acpi_link_adr *adr_link,
+			   int *dai_id, int *dai_num, int *codec_num,
+			   bool *group_generated)
+{
+	const struct snd_soc_acpi_adr_device *adr_d;
+	const struct snd_soc_acpi_link_adr *adr_next;
+	int group_id;
+	int index = 0;
+
+	*codec_num = adr_link->num_adr;
+	adr_d = adr_link->adr_d;
+
+	dai_id[index++] = ffs(adr_link->mask) - 1;
+	if (!adr_d->endpoints->aggregated) {
+		*dai_num = index;
+		return;
+	}
+
+	group_id = adr_d->endpoints->group_id;
+
+	/* gather other link ID of slaves in the same group */
+	for (adr_next = adr_link + 1; adr_next->num_adr;  adr_next++) {
```

Comment:
@RanderWang is the termination criterion "adr_next->num_adr" correct?

Reply:
yes, the last entry is empty, .e.g.
````
static const struct snd_soc_acpi_link_adr tgl_rvp[] = {
	{
		.mask = BIT(0),
		.num_adr = ARRAY_SIZE(rt711_0_adr),
		.adr_d = rt711_0_adr,
	},
	{
		.mask = BIT(1),
		.num_adr = ARRAY_SIZE(rt1308_1_adr),
		.adr_d = rt1308_1_adr,
	},
	{}
};
````

Reply:
ahh I see. I comment would be helpful then.